### DISCUSSION TITLE
problem setting proxy with seleniumbase SB

### DISCUSSION THREAD

**Discussion by sabermand** [seleniumbase/SeleniumBase#2550] [Q&A] [answered]
I've been using the following code snippet and SB from seleniumbase to pass turnstile and navigate through a website:

```
with SB(uc=True, demo=True) as sb:
    try:
        sb.driver.get(website1)
        for i in range(1, 3):
            if sb.is_text_visible("text"):
                print("Passed turnstile.")
                break
            else:
                print(f"Trial {i}: not passed yet")
        sb.get_new_driver(undetectable=True)
        sb.driver.get(website1)
    except Exception as e:
        print(f"An error occurred: {e}")

```
Now, I need to set up a rotating proxy for the driver, but I can't seem to find a solution. 
The sb.driver.get method does not support a proxy attribute, and the sb instance does not have a SB.config.proxy_string attribute.

I've tried using `sb.get_new_driver(undetectable=True, proxy='user:pass@ip:port')`, but it fails to load properly and gives a warning (see attached photo). 

How can I set up a proxy for this driver properly?"

![warning](https://github.com/seleniumbase/SeleniumBase/assets/30475004/d060f2d8-dc35-45b3-877e-7cc7b135b2bf)

**Comment by mdmintz** [comment, marked as answer]
That warning message about the `unsupported command-line flag` won't impact the result.
If you pass a valid `proxy` to `SB()`, `Driver()` or `get_new_driver()`, then it will work.

**Reply by sabermand** [reply]
could you please show how can i pass a proxy to SB() and also driver() ? I tried different methods but i get the attribute error. 


get_new_driver() accepts the proxy but fails to pass the turnstile. (keeps checking and refreshing).
```
sb.get_new_driver(undetectable=True, proxy='user:pass@ip:port')
sb.driver.get(website1)
```

I tested the proxy and it is valid with no issues. 
furthermore, I tried to pass the proxy to the system.

```
try:
    key = winreg.OpenKey(winreg.HKEY_CURRENT_USER, reg_path, 0, winreg.KEY_WRITE)
    winreg.SetValueEx(key, reg_key, 0, winreg.REG_SZ, proxy_server)
    winreg.SetValueEx(key, "ProxyEnable", 0, winreg.REG_DWORD, 1)
    winreg.SetValueEx(key, "ProxyUser", 0, winreg.REG_SZ, username)
    winreg.SetValueEx(key, "ProxyPass", 0, winreg.REG_SZ, password)
    winreg.CloseKey(key)
    print("Proxy settings applied successfully.")
```
it works on all apps with no problem. but SB() requires for user pass again. attached photo
![Picture2](https://github.com/seleniumbase/SeleniumBase/assets/30475004/1bbb9307-6931-4997-9fbe-46214934bd99)

could you give me some guidance here too? how can use SB to fill the user pass here. as they don't have selector

**Reply by mdmintz** [reply]
`sb.get_new_driver(undetectable=True, proxy="user:pass@ip:port")` is the correct format.
The proxy may have already been identified as a bot-traffic source, in which case it could be blocked based on the IP.
